AccountingServer.BLL/Util/Exchange.cs

```diff
@@ -150,4 +191,112 @@ private double Invoke(DateTime date, string from, string to)
             return rate;
         }
     }
-}
+
+    /// <summary>
+    ///     利用coinmartketcap.com查询实时汇率
+    /// </summary>
+    internal class CoinMarketCapExchange : ExchangeApi
+    {
+        /// <summary>
+        ///     缓存
+        /// </summary>
+        private readonly Dictionary<string, (DateTime, double)> m_Cache =
+            new Dictionary<string, (DateTime, double)>();
+
+        private readonly ReaderWriterLockSlim m_Lock = new ReaderWriterLockSlim();
+
+        /// <inheritdoc />
+        protected override double From(DateTime date, string target) => Invoke(date, target, BaseCurrency.Now);
+
+        /// <inheritdoc />
+        protected override double To(DateTime date, string target) => Invoke(date, BaseCurrency.Now, target);
+
+        /// <summary>
+        ///     调用查询接口
+        /// </summary>
+        /// <param name="date">日期</param>
+        /// <param name="from">基准</param>
+        /// <param name="to">目标</param>
+        /// <returns>汇率</returns>
+        private double Invoke(DateTime date, string from, string to)
+        {
+            if (from == to)
+                return 1;
+
+            var endpoint = date > DateTime.UtcNow
+                ? "latest"
+                : throw new ApplicationException("Historical rate unavailable.");
```

Comment:
https://coinmarketcap.com/api/documentation/v1/#operation/getV1ExchangeQuotesHistorical

Reply:
If it is paid plan we may just use latest rate...

Reply:
yep. we can stick on current rate then...
let's stick around alternatives, if possible in the future.

Reply:
or, ANOTHER TAR PIT: a reinvented currency conversion part?